mono/utils/mono-threads-posix.c

```diff
@@ -336,6 +336,16 @@ mono_memory_barrier_process_wide (void)
 	g_assert (status == 0);
 }
 
+gint32
+mono_native_thread_processor_id_get (void)
+{
+#ifdef HAVE_SCHED_GETCPU
+	return sched_getcpu ();
+#else
+	return -1;
+#endif
+}
```

Comment:
This is following the implementation in CoreCLR: 
https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/coreclr/src/pal/src/misc/sysinfo.cpp#L477-L487

Reply:
And it's worth noting we already have the `sched_getcpu` check in configure (even if we don't use it), which is why I didn't have to add it.